temporalcli/commands.workflow.go

```diff
@@ -110,8 +125,57 @@ func (c *TemporalWorkflowQueryCommand) run(cctx *CommandContext, args []string)
 	return cctx.Printer.PrintStructured(output, printer.StructuredOptions{})
 }
 
-func (*TemporalWorkflowResetCommand) run(*CommandContext, []string) error {
-	return fmt.Errorf("TODO")
+func (c *TemporalWorkflowResetCommand) run(cctx *CommandContext, _ []string) error {
+	if c.Type.Value == "" && c.EventId <= 0 {
+		return errors.New("must specify either valid event id or reset type")
+	}
+	cl, err := c.Parent.ClientOptions.dialClient(cctx)
+	if err != nil {
+		return err
+	}
+	defer cl.Close()
+
+	resetBaseRunID := c.RunId
+	eventID := int64(c.EventId)
+	if c.Type.Value != "" {
+		resetBaseRunID, eventID, err = getResetEventIDByType(cctx, c.Type.Value, c.Parent.Namespace, c.WorkflowId, c.RunId, cl.WorkflowService())
+		if err != nil {
+			return fmt.Errorf("getting reset event ID by type failed: %w", err)
+		}
+	}
+	reapplyType := enums.RESET_REAPPLY_TYPE_SIGNAL
+	if c.ReapplyType.Value != "All" {
+		reapplyType, err = enums.ResetReapplyTypeFromString(c.ReapplyType.Value)
+		if err != nil {
+			return err
+		}
+	}
+
+	if eventID > 0 {
+		cctx.Printer.Printlnf("Resetting workflow %s to event ID %d", c.WorkflowId, eventID)
```

Comment:
What does it mean for event ID to be 0? Should we always say something is being done to text-based output users instead of the possibility of no output?

Reply:
I had added that to debug some tests, but it shouldn't be possible for the eventID to be zero.

The old CLI didn't announce what event ID we were resetting to but it may be useful to know what eventID the CLI has decided to use when --type is specified